rules/always-return.js

```diff
@@ -244,6 +282,23 @@ module.exports = {
           return
         }
 
+        if (ignoreAssignmentVariable.length && isLastCallback(node)) {
+          let hasIgnoredAssignment = false
+
+          //istanbul ignore else
+          if (node.body?.type === 'BlockStatement') {
+            node.body.body.forEach((statement) => {
+              if (isIgnoredAssignment(statement, ignoreAssignmentVariable)) {
+                hasIgnoredAssignment = true
+              }
+            })
```

Comment:
Instead of the `hasIgnoredAssignment` variable, how about just:

```js
if (node.body.body.some((statement) => {
  return isIgnoredAssignment(statement, ignoreAssignmentVariable);
})) {
  return;
}
```